lib/engine/part/path.rb

```diff
@@ -126,10 +128,33 @@ def walk(skip: nil, jskip: nil, visited: nil, visited_edges: {}, on: nil, chain:
         if chain
           chained = chain + [self]
           yield chained if chain.empty? ? @nodes.size == 2 : @nodes.any?
+        elsif path_tree
+          path_chained = path_chain + [self]
+          if @nodes.any? { |n| n != current_node }
+            path_chained_id = path_chained.map { |p| "#{p.hex.id}-#{p.index}" }.join(':')
+            return if path_tree[path_chained_id]
```

Comment:
you don't need to do this, you can just store the paths in the path tree?

Reply:
No you right. I had a sort before on the id so it didnt matter which why you came from into the paths. This only works for Lawson tiles. But just storing the path_chained will work.